rafthttp/streamer.go

```diff
@@ -153,23 +155,24 @@ type WriteFlusher interface {
 
 // TODO: replace fs with stream stats
 type streamWriter struct {
+	w    WriteFlusher
 	to   types.ID
 	term uint64
-	fs   *stats.FollowerStats
+	mg   *metrics.Group
 	q    chan []raftpb.Entry
 	done chan struct{}
 }
 
-// newStreamServer starts and returns a new started stream server.
+// newStreamServer starts and returns a new unstarted stream server.
 // The caller should call stop when finished, to shut it down.
 func newStreamWriter(w WriteFlusher, to types.ID, term uint64) *streamWriter {
 	s := &streamWriter{
+		w:    w,
 		to:   to,
 		term: term,
 		q:    make(chan []raftpb.Entry, streamBufSize),
 		done: make(chan struct{}),
 	}
-	go s.handle(w)
```

Comment:
streamWriter needs to be set metricsGroup before handle goroutine starts(or there will be race), while streamWriter is inited in http handler, who doesn't know about the metricsGroup to use.